packages/contract/src/contract-factory.ts

```diff
@@ -154,18 +198,141 @@ export default class ContractFactory {
 
+    const transactionResponse = await account.sendTransaction(transactionRequest);
+
+    const waitForResult = async () => {
+      const transactionResult = await transactionResponse.waitForResult<TransactionType.Create>();
+      const contract = new Contract(contractId, this.interface, account) as TContract;
+
+      return { contract, transactionResult };
+    };
+
+    return { waitForResult, contractId, transactionId: transactionResponse.id };
+  }
+
+  /**
+   * Deploy a contract with the specified options.
+   *
+   * @param deployContractOptions - Options for deploying the contract.
+   * @returns A promise that resolves to the deployed contract instance.
+   */
+  async deployContractLoader<TContract extends Contract = Contract>(
+    deployContractOptions: DeployContractOptions = {}
+  ): Promise<DeployContractResult<TContract>> {
     const account = this.getAccount();
+    const { consensusParameters } = account.provider.getChain();
+    const maxContractSize = consensusParameters.contractParameters.contractMaxSize.toNumber();
 
-    const transactionResponse = await account.sendTransaction(transactionRequest);
+    // Ensure the max contract size is byte aligned before chunking
+    if (maxContractSize % WORD_SIZE !== 0) {
+      throw new FuelError(
+        ErrorCode.CONTRACT_SIZE_EXCEEDS_LIMIT, // Todo: change error
+        `Max contract size of ${maxContractSize} bytes is not byte aligned.`
+      );
+    }
+
+    const { configurableConstants } = deployContractOptions;
+    if (configurableConstants) {
+      this.setConfigurableConstants(configurableConstants);
+    }
+
+    // Find the max chunk size (again, ensure is word aligned)
+    const baseBlobRequest = this.blobTransactionRequest({ blobBytecode: randomBytes(32) });
+    baseBlobRequest.fundWithFakeUtxos([], account.provider.getBaseAssetId());
+    const maxChunkSize = maxContractSize - baseBlobRequest.byteLength() - WORD_SIZE;
+    const padding = maxChunkSize % WORD_SIZE;
+    const chunkSize = padding + ((WORD_SIZE - (padding % WORD_SIZE)) % WORD_SIZE) + maxChunkSize;
+
+    // Chunk the bytecode
+    const chunks: ContractChunk[] = [];
+    for (let offset = 0, index = 0; offset < this.bytecode.length; offset += chunkSize, index++) {
+      const chunk = this.bytecode.slice(offset, offset + chunkSize);
+
+      // Chunks must be byte aligned
+      if (chunk.length % WORD_SIZE !== 0) {
+        const paddingLength = chunk.length % WORD_SIZE;
+        const paddedChunk = concat([chunk, new Uint8Array(paddingLength)]);
+        chunks.push({ id: index, size: paddedChunk.length, bytecode: paddedChunk });
+        // eslint-disable-next-line no-continue
+        continue;
+      }
+      chunks.push({ id: index, size: chunk.length, bytecode: chunk });
+    }
+
+    const uploadedBlobIds: string[] = [];
+
+    // Deploy the chunks as blob txs
+    for (const { id, bytecode } of chunks) {
```

Comment:
1. Could all of this chunk deployment be done via a single multicall?
2. Is there a way to warn the user that they can't deploy the contract because they don't have enough funds? I imagine that it would be straightforward if it was a multicall.

Reply:
1. Not sure but I'll investigate.
2. As above not sure re multicall, but not difficult to estimate blobs + create tx anyway, will implement.

Reply:
Good ideas to look into.

Reply:
> 1. Could all of this chunk deployment be done via a single multicall?

I think not, because the [`max_size`](https://github.com/FuelLabs/fuels-ts/blob/db/feat/blob-contract-deploys/.fuel-core/configs/chainConfig.json#L11) of a transaction is the same as [`contract_max_size`](https://github.com/FuelLabs/fuels-ts/blob/aac2745f25dfeb9ee8e1ebc83c11f2268ff94ef4/.fuel-core/configs/chainConfig.json#L31) which we're [using](https://github.com/FuelLabs/fuels-ts/blob/36358fa37ca2bd032c88e0baa02ddab732a682a3/packages/contract/src/contract-factory.ts#L202) to determine if we'll deploy as blobs or not. A multicall transaction with all the blobs combined in it would suffer the same problem a `TransactionType.Create` transaction has and which we're mitigating with this PR.

 > 2. Is there a way to warn the user that they can't deploy the full contract because they don't have enough funds to cover all the blobs and contract loader transactions? 

This is still valid and should be done. We don't want to spend the user's funds on blob transactions if they can't deploy the whole contract.

Reply:
Apologies I hadn't completely looking into multicall yet but thankyou for the clarification. Agreed on the second point, I'll implement that today.

Reply:
Point 2 has now been implemented.